pkgs/by-name/ig/ignition/package.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  gettext,
+  desktop-file-utils,
+  appstream,
+  blueprint-compiler,
+  pkg-config,
+  gjs,
+  gtk4,
+  wrapGAppsHook4,
+  libadwaita,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ignition";
+  version = "1.1.3";
+
+  src = fetchFromGitHub {
+    owner = "flattool";
+    repo = "ignition";
+    tag = finalAttrs.version;
+    hash = "sha256-XVBlwonMHb78XF6mpPYLJ68E5Tb+dFVFqNSsVCCS0xc=";
+  };
+
+  patches = [
+    # hack to make gjs properly find our binary
+    ./fix-effective-entrypoint-path.patch
+  ];
+
+  postPatch = ''
+    # upstream uses gjs from the build platform instead of the host platform
+    # it's not very important, as I don't even know if cross compilation works with GTK
+    substituteInPlace src/meson.build \
+      --replace-fail "find_program('gjs').full_path()" "'${lib.getExe gjs}'"
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+
+    appstream
+    blueprint-compiler
+    desktop-file-utils
+    gettext
+    gtk4
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  meta = {
+    description = "Manage your startup apps and scripts on Freedesktop Linux distros";
```

suggestion: I prefer the [summary description](https://github.com/flattool/ignition/blob/1.1.3/data/io.github.flattool.Ignition.metainfo.xml.in#L15)

```suggestion
    description = "Manage startup apps and scripts";
```